x/ibc-hooks/wasm_hook.go

```diff
@@ -117,7 +111,21 @@ func (h WasmHooks) OnRecvPacketOverride(im IBCMiddleware, ctx sdk.Context, packe
 		return osmoutils.NewEmitErrorAcknowledgement(ctx, types.ErrWasmError, err.Error())
 	}
 
-	fullAck := ContractAck{ContractResult: response.Data, IbcAck: ack.Acknowledgement()}
+	// Check if the contract is requesting for the ack to be async.
+	var asyncAckRequest types.OnRecvPacketAsyncAckResponse
+	err = json.Unmarshal(response.Data, &asyncAckRequest)
+	if err == nil {
```

Comment:
I'm guessing this design choice of returning nil was made because we want to maintain backward compatibility with other contracts that utilize a different response data structure. Is that correct?

Reply:
Discussed offline - that seems to be correct and contract developers often return nil in their responses so we don't want to force them to return a specific response type